python-threatexchange/threatexchange/exchanges/impl/techagainstterrorism_api.py

```diff
@@ -121,13 +140,13 @@ def _type_mapping() -> t.Dict[str, str]:
 
 
 def _get_delta_mapping(
-    record: t.Dict[str, str],
-) -> t.Tuple[t.Tuple[str, str], t.Optional[state.FetchedSignalMetadata]]:
+    record: api.TATHashListEntry,
+) -> t.Tuple[t.Tuple[str, str], state.FetchedSignalMetadata]:
 
     if not _is_compatible_signal_type(record):
-        return (("", ""), None)
-
-    type_str = _type_mapping().get(record["algorithm"])
+        return (("", ""), state.FetchedSignalMetadata())
 
+    type_str = _type_mapping().get(record.algorithm, "")
     metadata = state.FetchedSignalMetadata()
-    return ((type_str or "", record["hash_digest"]), metadata)
+
+    return ((type_str, record.hash_digest), metadata)
```

Comment:
blocking: How to handle deletions:

```suggestion
    return ((type_str, record.hash_digest), None if record.deleted else metadata)
```